chsdi/views/files.py

```diff
@@ -0,0 +1,242 @@
+
+#from chsdi.models.clientdata_dynamodb import get_table
+
+from boto.s3.connection import S3Connection
+from boto.s3.key import Key
+
+from pyramid.view import view_config, view_defaults
+import pyramid.httpexceptions as exc
+from pyramid.response import Response
+
+from chsdi.lib.decorators import requires_authorization
+
+
+def validate_kml_input():
+    def decorator(func):
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            if hasattr(self, 'request'):
+                request = self.request
+            else:
+                request = self
+
+            MAX_FILE_SIZE = 1024 * 1024 * 2
+
+            if request.content_type != 'application/vnd.google-earth.kml+xml':
+                raise exc.HTTPUnsupportedMediaType('Only KML file are accepted')
+            try:
+                data = request.body
+                if len(data) > MAX_FILE_SIZE:
+                    raise exc.HTTPRequestEntityTooLarge('File size exceed %s bytes' % MAX_FILE_SIZE)
+                p = xml.parsers.expat.ParserCreate()
+                p.Parse(data)
+            except Exception:
+                raise exc.HTTPUnsupportedMediaType('Only valid KML file are accepted')
+
+            return func(self, *args, **kwargs)
+        return wrapper
+    return decorator
+
+
+def _get_dynamodb_table():
+    PROFILE_NAME = 'Credentials'
+    config = ConfigParser.ConfigParser()
+    config.read(["/etc/boto.cfg"])
+
+    access_key = config.get(PROFILE_NAME, 'aws_access_key_id')
+    secret_key = config.get(PROFILE_NAME, 'aws_secret_access_key')
+
+    conn = connect_to_region('eu-west-1', aws_access_key_id=access_key,
+                             aws_secret_access_key=secret_key)
+
+    table = Table('geoadmin-file-storage', connection=conn)
+    #table = get_table(table_name='geoadmin-file-storage')
+
+    return table
+
+
+    # return get_table(table_name='geoadmin-file-storage')
+
+def _add_item(id, file_id=False):
+    table = _get_dynamodb_table()
+    try:
+        table.put_item(
+            data={
+                'adminId': id,
+                'fileId': file_id,
+                'timestamp': time.strftime('%Y-%m-%d %X', time.localtime())
+            }
+        )
+    except Exception as e:
+            raise exc.HTTPBadRequest('Error during put item %s' % e)
+    return True
+
+
+def _is_admin_id(adminId):
+    # TODO: search in database if it's an adminId
+    table = _get_dynamodb_table()
+    try:
+        table.get_item(adminId=str(adminId))
+    except ItemNotFound:
+        return False
+
+    return True
+
+
+def _get_file_id_from_admin_id(admin_id):
+    # TODO: get the file id from database
+    fileId = None
+    table = _get_dynamodb_table()
+    try:
+        item = table.get_item(adminId=str(admin_id))
+        fileId = item.get('fileId')
+    except Exception as e:
+        raise exc.HTTPBadRequest('The id %s doesn\'t exist. Error is: %s' % (id, e))
+    if fileId is None:
```

Comment:
Should never be the case. You only have *one* id, but you only dynamoDB knows what is an *fileId* and what is an *adminId*.